manager/controllers/app/plotter_controller.go

```diff
@@ -0,0 +1,246 @@
+		if err := r.Client.Status().Update(ctx, &plotter); err != nil {
+			return ctrl.Result{}, errors.WrapWithDetails(err, "failed to update plotter status", "status", plotter.Status)
+		}
+	}
+
+	log.Info("plotter reconcile cycle completed", "result", result)
+	return result, nil
+}
+
+// reconcileFinalizers reconciles finalizers for Plotter
+func (r *PlotterReconciler) reconcileFinalizers(plotter *app.Plotter) error {
+	// finalizer
+	finalizerName := r.Name + ".finalizer"
+	hasFinalizer := ctrlutil.ContainsFinalizer(plotter, finalizerName)
+
+	// If the object has a scheduled deletion time, delete it and its associated resources
+	if !plotter.DeletionTimestamp.IsZero() {
+		// The object is being deleted
+		if hasFinalizer { // Finalizer was created when the object was created
+			// the finalizer is present - delete the allocated resources
+
+			for cluster, blueprint := range plotter.Status.Blueprints {
+				// TODO Check namespace deletion. Some finalizers leave namespaces in terminating state
+				err := r.ClusterManager.DeleteBlueprint(cluster, blueprint.Namespace, blueprint.Name)
+				if err != nil {
+					return err
+				}
+			}
+
+			// remove the finalizer from the list and update it, because it needs to be deleted together with the object
+			ctrlutil.RemoveFinalizer(plotter, finalizerName)
+
+			if err := r.Update(context.Background(), plotter); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+	// Make sure this CRD instance has a finalizer
+	if !hasFinalizer {
+		ctrlutil.AddFinalizer(plotter, finalizerName)
+		if err := r.Update(context.Background(), plotter); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *PlotterReconciler) reconcile(plotter *app.Plotter) (ctrl.Result, error) {
+	isInitialReconcile := false
+	if plotter.Status.Blueprints == nil {
+		plotter.Status.Blueprints = make(map[string]app.MetaBlueprint)
+		isInitialReconcile = true
+	}
+
+	// Reconciliation loop per cluster
+	isReady := true
+	for cluster, blueprintSpec := range plotter.Spec.Blueprints {
+		r.Log.V(1).Info("Handling cluster " + cluster)
+		if blueprint, exists := plotter.Status.Blueprints[cluster]; exists {
+			r.Log.V(2).Info("Found status for cluster " + cluster)
+
+			remoteBlueprint, err := r.ClusterManager.GetBlueprint(cluster, blueprint.Namespace, blueprint.Name)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			if remoteBlueprint == nil {
+				r.Log.Info("Could not yet find remote blueprint")
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+			}
+
+			if !reflect.DeepEqual(blueprintSpec, remoteBlueprint.Spec) {
+				r.Log.V(1).Info("Blueprint specs differ...")
+				remoteBlueprint.Spec = blueprintSpec
+				remoteBlueprint.ObjectMeta.Annotations = map[string]string(nil) // reset annotations
+				err := r.ClusterManager.UpdateBlueprint(cluster, remoteBlueprint)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
Not very effective to requeue inside the iteration (in case we have multiple blueprints)